app/move/controllers/create/assessment.js

```diff
@@ -1,15 +1,47 @@
-const { flatten, values } = require('lodash')
+const { forEach, flatten, values, mapValues } = require('lodash')
 
 const CreateBaseController = require('./base')
 const fieldHelpers = require('../../../../common/helpers/field')
+const createFields = require('../../fields/create')
+const referenceDataService = require('../../../../common/services/reference-data')
 
 class AssessmentController extends CreateBaseController {
   async configure(req, res, next) {
     try {
-      const { fields } = req.form.options
-      req.form.options.fields = await fieldHelpers.populateAssessmentQuestions(
-        fields
+      const { fields, assessmentCategory } = req.form.options
+      const questions = await referenceDataService.getAssessmentQuestions(
+        assessmentCategory
       )
+
+      const implicitField = createFields.assessmentCategory(assessmentCategory)
+      implicitField.items = questions
+        .filter(question => {
```

Comment:
Maybe move this to a field helper too. Will make it easier to test in isolation.

It could also then be reduced to this:

```js
...
.filter(fieldHelpers.filterExplicitFields(fields))
...
```

```js
function filterExplicitFields(fields) {
  return { key } => {
    const key = question.key
    const field = fields[key]

    return field && !field.explicit
  }
}
```

Reply:
I would probably try and extract all of this logic into a field helper too. That way it's not tied specifically to this controller and you can test each method in a simpler way.

It would be nice to reduce the controller to just one call to a helper that returns an unmutated set of fields, like before.